src/App.tsx

```diff
@@ -77,80 +188,75 @@ function App() {
 
   const getSection = () => {
     // TODO: should never happen
-    if (!queryParamsResults.success) {
+    if (!queryParamsResults.success || !appState) {
       return
     }
-    if (step === Steps.One) {
-      return (
-        <SignInScreen
-          onError={onError}
-          onNext={setStep}
-          params={queryParamsResults.data}
-          setLinkedAccount={setLinkedAccount}
-        />
-      )
-    }
-    if (step === Steps.Two) {
-      return (
-        <PaymentInfoScreen
-          onError={onError}
-          onNext={setStep}
-          params={queryParamsResults.data}
-          setLinkedAccount={setLinkedAccount}
-        />
-      )
-    }
-    if (step === Steps.Three && linkedAccount) {
-      return (
-        <ReviewScreen
-          onError={onError}
-          onNext={setStep}
-          params={queryParamsResults.data}
-          linkedAccount={linkedAccount}
-          setTransferResponse={setTransferResponse}
-        />
-      )
-    }
-
-    if (
-      step === Steps.Four &&
-      transferResponse &&
-      'userActionDetails' in transferResponse &&
-      transferResponse.userActionDetails &&
-      queryParamsResults.data.transferType === TransferType.TransferIn
-    ) {
-      return (
-        <UserActionDetails
-          onNext={setStep}
-          userActionDetails={transferResponse.userActionDetails}
-          fiatAmount={queryParamsResults.data.fiatAmount}
-          fiatType={queryParamsResults.data.fiatType}
-          providerId={queryParamsResults.data.providerId}
-        />
-      )
-    }
-
-    if (
-      step === Steps.Four &&
-      transferResponse &&
-      queryParamsResults.data.transferType === TransferType.TransferOut
-    ) {
-      return (
-        <SendCrypto
-          onNext={setStep}
-          onError={onError}
-          transferAddress={transferResponse.transferAddress}
-          cryptoAmount={queryParamsResults.data.cryptoAmount}
-          cryptoType={queryParamsResults.data.cryptoType}
-          providerId={queryParamsResults.data.providerId}
-        />
-      )
-    }
-
-    if (step === Steps.Five) {
-      // TODO: Actually figure this out, and have sensible defaults like we do in the wallet
-      const settlementTime = '1 - 3 days'
-      return <DoneSection settlementTime={settlementTime} />
+    switch (appState.currentScreen) {
+      case Screens.SignInScreen: {
+        return (
+          <SignInScreen
+            onError={onError}
+            onNext={() => setFinishedSignIn(true)}
+            params={queryParamsResults.data}
+            setLinkedAccount={setLinkedAccount}
+          />
+        )
+      }
+      case Screens.PaymentInfoScreen: {
+        return (
+          <PaymentInfoScreen
+            onError={onError}
+            params={queryParamsResults.data}
+            setLinkedAccount={setLinkedAccount}
+          />
+        )
+      }
+      case Screens.ReviewScreen: {
+        if (!linkedAccount) return
```

Comment:
seeing this and similar return undefined's for later screens makes me wonder if we should add a loading indicator placeholder at some point. we can do that as a followup if we end up seeing blank sections here a lot and it impacts the ux